lib/broadway_rabbitmq/producer.ex

```diff
@@ -433,20 +433,34 @@ defmodule BroadwayRabbitMQ.Producer do
   end
 
   defp ack_messages(messages, channel, kind) do
-    Enum.each(messages, fn msg ->
+    Enum.flat_map(messages, fn msg ->
       {_module, _channel, ack_data} = msg.acknowledger
 
-      try do
-        case kind do
-          :successful -> apply_ack_func(ack_data.on_success, ack_data, channel)
-          :failed -> apply_ack_func(ack_data.on_failure, ack_data, channel)
-        end
-      catch
-        kind, reason ->
-          formatted_error = Exception.format(kind, reason, __STACKTRACE__)
-          Logger.error("Could not ack/reject message: #{formatted_error}")
+      case apply_ack_func(kind, ack_data, channel) do
+        {:error, reason} -> [{msg, reason}]
+        _success -> []
       end
     end)
+    |> case do
+      [] ->
+        :ok
+
+      [{msg, reason} | _other_failures] ->
+        raise RuntimeError, """
+        could not ack/reject message
+
+        Message: #{inspect(msg)}
+        Reason: #{inspect(reason)}
+        """
```

Comment:
@josevalim with this though, since we don't log per-message failures, we'll only surface that we failed one message, regardless of how many we failed. We won't have any information about the other messages we failed to ack. Maybe we can raise an error that shows *all* messages, or maybe we can log inside the `flat_map` there, and then raise here like we're doing?

Reply:
Logging + raising OR raising with all error messages work for me. @joeLepper, thoughts?

Reply:
I'm going to log and raise as I think trying to raise with all of that information in a single message will be a bit verbose. I'll push something up ASAP.